Document data retention/data privacy (#413)
TheThingsIndustries/lorawan-stack-docs: doc/content/reference/data-retention/_index.md

---

> title: "Data Retention and Privacy"

Reviewer: This is strictly about TTSC, right? Should we clarify that?

description: ""
weight:
---

This reference contains information about how long {{% tts %}} persists device data. For inquiries about data privacy or data security, read on, or [contact us](cloud@thethingsindustries.com).

<!--more-->

{{% tts %}} Network Server stores

- Recent uplinks as part of the MAC state, in a Redis database

{{% tts %}} Application Server stores

- Recent uplink application payloads, in a Redis database

{{% tts %}} Cloud runs on an isolated subnet, and the databases can only be accessed from within the instance.

{{% tts %}} uses your Network Key to decrypt LoRaWAN packets - these can only be decrypted at the Network Server, not by gateways or anywhere else along the IP route they travel.

> {{% tts %}} uses your Application Key if provided to decrypt the application payload, which is available in the [JSON message]({{< ref "reference/data-formats" >}}). It is also possible to skip payload decryption in {{% tts %}}, and perform decryption later in your application pipeline. Select **Skip payload encryption/decryption** when registering a device to do so.

Reviewer: ```suggestion
{{% tts %}} uses the LoRaWAN AppSKey to decrypt the application payload, which is available in the [JSON message]({{< ref "reference/data-formats" >}}). It is also possible to skip payload decryption in {{% tts %}}, and perform decryption later in your application pipeline. Select **Skip payload encryption/decryption** when registering a device to do so.
```

{{< figure src="skip-encryption.png" >}}

> The Things Industries will never read application payloads, even if you provide your Application Key.

Reviewer: ```suggestion
The Things Industries will never read application payloads, even if the AppSKey is configured.
```

Our services do process unencrypted payload. What do you mean with "The Things Industries", the people? Support and operations will sometimes need to access data streams on the application level.

Author: https://github.com/TheThingsIndustries/lorawan-stack-support/issues/402#issuecomment-825763564

Author: should we say "The Things Industries personnel will never access your application payloads, unless given permission to do so to provide support."

---
title: "Data Retention and Privacy"
description: ""
weight:
---

This reference contains information about what access administrators and users have to data, and how long {{% tts %}} persists device data. For inquiries about data privacy or data security, read on, or [contact The Things Industries](cloud@thethingsindustries.com).

<!--more-->

## Important Terms

To understand how data is secured in {{% tts %}} it is necessary to understand some important roles:

> **Infrastructure Owners**: Those with access to the hardware {{% tts %}} runs on. For {{% tts %}} Cloud and Community Edition, this is The Things Industries. For Enterprise and Open Source deployments, this is you, the customer. Formally, the **Infrastructure Owner** has access to all data stored in a database on the hardware. In practice, The Things Industries does not use this to access data in {{% tts %}} Cloud or Community Edition deployments.

Reviewer: ```suggestion
**Infrastructure Owners**: Those with access to the hardware {{% tts %}} runs on. For {{% tts %}} Cloud, Dedicated Cloud and Community Edition, this is The Things Industries. {{% tts %}} Enterprise and Open Source deployments are managed by the customer and user respectively; The Things Industries does not have any infrastructural access. Technically, **Infrastructure Owners** have access to all data stored in databases or persistent volumes. In practice, The Things Industries uses this direct data access to {{% tts %}} Cloud or Community Edition deployments only for backups.
```

> **Network Administrators**: Users with `admin` access in a tenant. **Network Administrators** have access to all entities in a tenant, including device secrets. In multi-tenant environments, tenants are completely isolated, and **Network Administrators** of one tenant have **no access** to other tenants.

Reviewer: ```suggestion
**Network Administrators**: Users with administrative access in a tenant. **Network Administrators** have access to all entities in a tenant, including device secrets. In multi-tenant environments, tenants are completely isolated, and **Network Administrators** of one tenant have **no access** to other tenants.
```

> **Users and Organizations**: Users without `admin` access, and organizations, can access entities which they have [collaborator rights]({{< ref "getting-started/user-management" >}}) over.

Reviewer: ```suggestion
**Users and Organizations**: Users without administrative access, and organizations, can access entities which they have [collaborator rights]({{< ref "getting-started/user-management" >}}) over.
```

> 

Reviewer: Add section support engineers? I think it's good to mention that TTI support engineers do not have infrastructural access, and customers should allow them explicitly and temporarily to access applications and gateways in their tenant.

> ## Cloud Deployments

Reviewer: ```suggestion
## {{% tts %}} Cloud and Dedicated Cloud
```

The following information applies to {{% tts %}} deployments hosted by The Things Industries, specifically {{% tts %}} Cloud and Dedicated Cloud.

{{% tts %}} deployments hosted on customer hardware, specifically {{% tts %}} Enterprise, are completely managed by the customer and can be configured to meet more specific data privacy requirements.

> ## Persisted Data

Reviewer: ```suggestion
### Persisted Data
```

On {{% tts %}} Cloud and Dedicated Cloud, the Network Server stores

- Recent uplinks as part of the MAC state, in a Redis database

The Application Server stores

> - Recent uplink application payloads, in a Redis database

Reviewer: Is this really necessary to mention?

If so, we should also mention that Application Server stores upstream messages through the Storage Integration (+ link) if that is enabled. Also mention the data retention here.

> {{% tts %}} Cloud runs on an isolated subnet, and the databases can only be accessed from within the instance.

Reviewer: ```suggestion
{{% tts %}} Cloud runs in isolated subnets. Databases can only be accessed from within the subnet and are not publicly accessible.
```

> {{% tts %}} uses your Network Key to decrypt LoRaWAN packets - these can only be decrypted at the Network Server, not by gateways or anywhere else along the IP route they travel.

Reviewer: ```suggestion
{{% tts %}} uses your NwkSKey (NwkSEncKey in LoRaWAN 1.1) to decrypt LoRaWAN network-level payload. These session keys are only known to the Network Server and are never exposed downstream (to gateways) or upstream (to applications).
```

{{% tts %}} uses the LoRaWAN AppSKey to decrypt the application payload, which is available in the [JSON message]({{< ref "reference/data-formats" >}}). It is also possible to skip payload decryption in {{% tts %}}, and perform decryption later in your application pipeline. Select **Skip payload encryption/decryption** when registering a device to do so.

{{< figure src="skip-encryption.png" >}}

> The Things Industries personnel will never access your application payloads, even if the AppSKey is configured, unless given permission to do so to provide support.

Reviewer: I think this line can be replaced by mentioning above what support engineers can and cannot do
